tests/test_elegant_conversion.py

```diff
@@ -32,14 +33,42 @@ def test_fodo():
     assert [element.name for element in converted.elements] == [
         element.name for element in correct_lattice.elements
     ]
-    assert converted.q1.length == correct_lattice.q1.length
-    assert converted.q1.k1 == correct_lattice.q1.k1
-    assert converted.q2.length == correct_lattice.q2.length
-    assert converted.q2.k1 == correct_lattice.q2.k1
-    assert [d.length for d in converted.d1] == [d.length for d in correct_lattice.d1]
-    assert converted.d2.length == correct_lattice.d2.length
-    assert converted.s1.length == correct_lattice.s1.length
-    assert converted.s1.e1 == correct_lattice.s1.e1
+    assert torch.isclose(converted.q1.length, correct_lattice.q1.length)
+    assert torch.isclose(converted.q1.k1, correct_lattice.q1.k1)
+    assert torch.isclose(converted.q2.length, correct_lattice.q2.length)
+    assert torch.isclose(converted.q2.k1, correct_lattice.q2.k1)
+    for i in range(2):
+        assert torch.isclose(converted.d1[i].length, correct_lattice.d1[i].length)
+    assert torch.isclose(converted.d2.length, correct_lattice.d2.length)
+    assert torch.isclose(converted.s1.length, correct_lattice.s1.length)
+    assert torch.isclose(converted.s1.e1, correct_lattice.s1.e1)
```

Comment:
I feel like this should be no loop and use `torch.allclose`.

Reply:
I`m not sure how `torch.allclose` would work here, but maybe there is some trick I`m missing. The problem is that we have two elements with name `d1` and therefore two separate `length` tensors, not one multidimensional tensor.

Reply:
Oh I see ... I misunderstood the that line 🙃